### ISSUE TITLE
Rancher HA upgrade fails when antiAffinity is set to required

### ISSUE THREAD

**Issue by dnoland1** [rancher/rancher#24518]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
1. Launch a Rancher HA environment with 3 nodes, make sure helm chart option `antiAffinity` is set to `required` (default is `preferred`). Deploy version 2.3.2.
2. Upgrade to version 2.3.3

**Result:**
The new rancher pods fail to deploy. On the UI, you'll see the error "0/3 nodes are available: 3 node(s) didn't match pod affinity/anti-affinity, 3 node(s) didn't satisfy existing pods anti-affinity rules."

**Other details that may be helpful:**
This is happening because of the deployment yaml for Rancher. It sets the following antiaffinity rule:
```
      affinity:
        podAntiAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
          - labelSelector:
              matchExpressions:
              - key: app
                operator: In
                values:
                - rancher
            topologyKey: kubernetes.io/hostname
```
However, the rollout strategy for maxUnavailable is:
```
  strategy:
    rollingUpdate:
      maxSurge: 25%
      maxUnavailable: 25%
    type: RollingUpdate
```
Since maxUnavailable is rounded down, it's effectively 0, meaning Kubernetes cannot stop a pod and go to 2 replicas to allow a new pod to deploy. Due to the antiaffinity required rule, it cannot go to 4 replicas. Therefore, the upgrade is stuck.

IMHO, values for rolling update strategy should be maxSurge: 1, maxUnavailable: 1. Tried setting maxUnavailable to 1 manually and this fixes the problem. You can also work around the issue by temporarily setting replicas to 2 (from 3) in the UI, letting it upgrade, then setting replicas back to 3.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
v2.3.2 and v2.3.3
- Installation option (single install/HA):
HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
NA
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
t3a.medium on AWS
- Kubernetes version (use `kubectl version`):
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-11-14T04:24:29Z", GoVersion:"go1.12.13", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-11-13T11:13:49Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
(paste the output here)
```

**Comment by jiaqiluo** [comment]
The bug is reproduced with the upgrade path from v2.3.2 to v2.3.3

Install Rancher v2.3.2  
```
helm install rancher rancher-latest/rancher \
--namespace cattle-system \
--version v2.3.2 \
--set hostname=jiaqi-ha.qa.rancher.space \
--set antiAffinity="required"
```

Then upgrade to v2.3.3
```
helm upgrade rancher rancher-latest/rancher \
--namespace cattle-system \
--version v2.3.3 \
--set hostname=jiaqi-ha.qa.rancher.space \
--set antiAffinity="required"
```

Results:
- new pod cannot be deployed and the upgrade is paused 

<img width="1414" alt="Screen Shot 2019-12-11 at 6 03 01 PM" src="https://user-images.githubusercontent.com/6218999/70673763-f01ae300-1c40-11ea-9ec6-b06e97a1fc64.png">

**Comment by jiaqiluo** [comment]
The bug fix is validated with Rancher:v2.3.4-rc3

Install Rancher v2.3.2  
```
helm install rancher rancher-latest/rancher \
--namespace cattle-system \
--version v2.3.2 \
--set hostname=jiaqi-ha.qa.rancher.space \
--set antiAffinity="required"
```

Then upgrade to v2.3.4-rc3
```
helm upgrade rancher rancher-latest/rancher \
--namespace cattle-system \
--version v2.3.4-rc3 \
--set hostname=jiaqi-ha.qa.rancher.space \
--set antiAffinity="required"
```

Results:
- upgrade succeeds, all pods are recreated and active 

<img width="1474" alt="Screen Shot 2019-12-12 at 4 46 11 PM" src="https://user-images.githubusercontent.com/6218999/70758476-5ca8e700-1d00-11ea-99a3-9876ae307b23.png">

**Comment by jiaqiluo** [comment]
Wait for a v2.4-preview to validaste

**Comment by jiaqiluo** [comment]
The bug fix is validated with Rancher:v2.4.0-alpha1

Install Rancher v2.3.2  
```
helm install rancher rancher-latest/rancher \
--namespace cattle-system \
--version v2.3.2 \
--set hostname=jiaqi-ha.qa.rancher.space \
--set antiAffinity="required"
```

Then upgrade to v2.4.0-alpha1
```
helm upgrade rancher rancher-alpha/rancher \
--namespace cattle-system \
--version 2.4.0-alpha1 \
--set hostname=jiaqi-ha.qa.rancher.space \
--set antiAffinity="required"
```

Results:
- upgrade succeeds, all pods are recreated and active
